api/pom.xml

```diff
@@ -373,8 +392,8 @@
                                     <version>3.5.4</version>
                                 </requireMavenVersion>
                                 <requireJavaVersion>
-                                    <version>[1.8.0,)</version>
-                                    <message>You need JDK8 or higher</message>
+                                    <version>[9,)</version>
```

Comment:
This updates the Java compliation version, which means Java 8 is no longer working with the new EJB release. I think we need multi release to support Java 8 as well, same as other spec does.

Reply:
I haven't followed this issue closely, so I may not have the complete picture. But from looking at the discussion so far, with `maven.compiler.target` set to `1.8`, the produced jakarta.ejb-api-4.0.1.jar should be able to work with Java 8. Granted, users won't be able to build the api from source with jdk 8, but that can worked around by using jdk 9+.

Having to create a multi release for adding module info seems not worth it.

Reply:
I think we should follow the EE 10 module-info approach to be consistent.

Reply:
This PR follows the Jakarta EE 10 module-info approach documented here :
https://github.com/eclipse-ee4j/jakartaee-platform/wiki/Modularized-Jars

_If your spec is not updating the API and producing a service release that needs to support Java SE 8, run two passes of the compiler-plugin to produce the SE 8 based content, followed by a pass to produce the SE 9 based module-info.class._

PR #146 is similar, but opposite. PR #146 first compiles all classes with Java 11, then on a second pass re-compiles all classes again, except for module-info using Java 8.  While the output API JAR is nearly the same, the disadvantage is that PR #146 compiles most of the class files twice, discarding the Java 11 version.

This PR #141 follows the Jakarta EE 10 recommendation, and compiles all except module-info first using Java 8, and then on the second pass only compiles module-info using Java 9.

Neither PR creates a multi-release JAR. That is, neither places any alternate version classes under the META-INF directory.  Both PRs just add `module-info.class` at the root of the JAR.  However, this PR, #141 just does it more efficiently and follows the Jakarta EE 10 recommendation more accurately.

Reply:
I would like to add one more point, my PR keeps **transaction-api** at **2.0.0** as it already has automatic module name defined as `jakarta.transaction`. When this is upgraded to 2.0.1, it requires all the changes @starksm64 has done.
1. Add cdi-api as provided dependency
2. JDK11 is required to compile module-info because of `java.transaction.xa` module (available since JDK11).

Reply:
Thank you @tkburroughs for your explaination! Undestood now.